spark/src/main/java/com/netflix/iceberg/spark/source/IcebergSource.java

```diff
@@ -109,10 +113,19 @@ protected SparkSession lazySparkSession() {
     return lazySpark;
   }
 
-  protected Configuration lazyConf() {
+  protected Configuration lazyBaseConf() {
     if (lazyConf == null) {
       this.lazyConf = lazySparkSession().sparkContext().hadoopConfiguration();
     }
     return lazyConf;
   }
+
+  protected Configuration mergeIcebergHadoopConfs(Configuration baseConf, Map<String, String> options) {
+    Configuration resolvedConf = new Configuration(baseConf);
+    options.keySet().stream()
+        .filter(key -> key.startsWith("iceberg.hadoop"))
```

Comment:
Why not pass all of the options? Most of them would simply be ignored, right?

Reply:
You don't want to set the Hadoop properties directly in `DataSourceOptions`, I think. Setting `fs.myscheme.impl` in `DataSourceOptions`, for example, doesn't lend itself to clear context that it would be applied as a Hadoop configuration. Prefixing makes it clear what these options would be used for. But when we prefix we have to strip the `iceberg.hadoop` prefix when passing down to the `Configuration` object.

This is a precedent set by Spark itself as well.

Reply:
That sounds fine to me.